OutOfSchool/OutOfSchool.WebApi/Services/Images/ImageDependentEntityImagesInteractionMediator.cs

```diff
@@ -21,152 +21,168 @@ namespace OutOfSchool.WebApi.Services.Images
+            if (string.IsNullOrEmpty(entity.CoverImageId))
             {
-                return new ImageRemovingResult
-                {
-                    MultipleKeyValueOperationResult = new MultipleKeyValueOperationResult { GeneralResultMessage = ImagesOperationErrorCode.NoGivenImagesError.GetResourceValue() },
-                };
+                throw new InvalidOperationException(@"Image id must be a non empty string");
             }
 
-            var entity = await GetEntityWithIncludedImages(entityId).ConfigureAwait(false);
-            if (entity == null)
+            var removingCoverImageResult = await ImageService.RemoveImageAsync(entity.CoverImageId).ConfigureAwait(false);
+
+            if (removingCoverImageResult.Succeeded)
             {
-                Logger.LogError($"Entity [id = {entityId}] was not got.");
-                return new ImageRemovingResult
-                {
-                    MultipleKeyValueOperationResult = new MultipleKeyValueOperationResult { GeneralResultMessage = ImagesOperationErrorCode.EntityNotFoundError.GetResourceValue() },
-                };
+                entity.CoverImageId = null;
             }
 
-            return await RemoveManyImagesProcessAsync(entity, imageIds).ConfigureAwait(false);
+            return removingCoverImageResult;
         }
 
-        /// <summary>
-        /// Gets the entity with included images that is required to be non-null.
-        /// </summary>
-        /// <param name="entityId">Entity id.</param>
-        /// <returns>The instance of <see cref="Task"/> that represents <c>TEntity</c>.</returns>
-        /// <exception cref="InvalidOperationException">When entity is null.</exception>
-        protected async Task<TEntity> GetRequiredEntityWithIncludedImages(TKey entityId)
+        /// <inheritdoc/>
+        public async Task<ImageChangingResult> ChangeCoverImageAsync(TEntity entity, string dtoImageId, IFormFile newImage)
         {
-            var entity = await GetEntityWithIncludedImages(entityId).ConfigureAwait(false);
-            return entity ?? throw new InvalidOperationException($"Unreal to find {nameof(TEntity)} with id {entityId}.");
-        }
+            _ = entity ?? throw new ArgumentNullException(nameof(entity));
 
-        /// <summary>
-        /// Gets the entity with included images.
-        /// </summary>
-        /// <param name="entityId">Entity id.</param>
-        /// <returns>The instance of <see cref="Task"/> that represents <c>TEntity</c>.</returns>
-        /// <exception cref="InvalidOperationException">When filter is not specified.</exception>
-        protected virtual async Task<TEntity> GetEntityWithIncludedImages(TKey entityId)
-        {
-            return (await Repository.GetByFilter(x => x.Id.Equals(entityId), nameof(IImageDependentEntity<TEntity>.Images)).ConfigureAwait(false)).FirstOrDefault();
+            if (string.Equals(dtoImageId, entity.CoverImageId, StringComparison.Ordinal) &&
```

Comment:
Please swap && operands. 
It makes sense to check `(!string.IsNullOrEmpty(dtoImageId) || newImage == null)` first and only then use more complicated `string.Equals`